dmriprep/utils/vectors.py

```diff
@@ -1,482 +1,346 @@
-    elif np.sum(np.invert(gtab.b0s_mask)) < 30:
-        slm = 'linear'
-        print('Fewer than 30 diffusion-weighted directions detected. To ensure adequate eddy current correction, eddy '
-              'will be run using the --slm=linear option')
-    else:
-        slm = 'none'
-    return slm
-
-
-def check_corruption(bvecs, bvals, B0_NORM_EPSILON):
-    """
-    Performs a variety of basic check to ensure bval/bvec files are not corrupted.
-
-    Parameters
-    ----------
-    bvecs : m x n 2d array
-        Raw b-vectors array.
-    bvals : 1d array
-        Raw b-values float array.
-    B0_NORM_EPSILON : float
-        Gradient threshold below which volumes and vectors are considered B0's.
-
-    Returns
-    -------
-    bvecs : m x n 2d array
-        Raw b-vectors array.
-    bvals : 1d array
-        Raw b-values float array.
-    """
-    # Correct any b0's in bvecs misstated as 10's.
-    bvecs[np.where(np.any(abs(bvecs) >= 10, axis=1) == True)] = [1, 0, 0]
-
-    # Check for bval-bvec discrepancy.
-    if len(bvecs[np.where(np.logical_and(bvals > B0_NORM_EPSILON, np.where(np.all(abs(bvecs)) == 0)))]) > 0:
-        raise ValueError("Corrupted bval/bvecs detected. Check to ensure volumes with a diffusion weighting are not "
-                         "listed as B0s in the bvecs.")
-
-    # Ensure that B0's in the bvecs are represented are zero floats.
-    bvecs[np.where(bvals < B0_NORM_EPSILON)] = float(0)
+    return pole
 
-    return bvecs, bvals
 
-
-def rescale_vectors(bvals, bvecs, bval_scaling, bmag, B0_NORM_EPSILON):
+def bvecs2ras(affine, bvecs, norm=True, bvec_norm_epsilon=0.2):
     """
-    Performs a variety of rescaling routines on the b vectors and values.
-
-    Parameters
-    ----------
-    bvals : 1d array
-        Raw b-values float array.
-    bvecs : m x n 2d array
-        Raw b-vectors array.
-    bval_scaling : bool
-        If True, then normalizes b-val by the square of the vector amplitude.
-    bmag : int
-        The order of magnitude to round the b-values.
-    B0_NORM_EPSILON : float
-        Gradient threshold below which volumes and vectors are considered B0's.
+    Convert b-vectors given in image coordinates to RAS+.
 
-    Returns
-    -------
-    bval_mag_norm : 1d array
-        Rescaled b-values float array.
-    bvec_rescaled : m x n 2d array
-        Rescaled b-vectors array.
-    """
-    from dipy.core.gradients import round_bvals
-    from dmriprep.utils.vectors import rescale_bvec, rescale_bval
-    # Rescale bvals by vector amplitudes.
-    if bval_scaling is True:
-        bvals = rescale_bval(bvals.astype('float'), bvecs.astype('float'), B0_NORM_EPSILON)
+    Examples
+    --------
+    >>> bvecs2ras(2.0 * np.eye(3), [(1.0, 0.0, 0.0), (-1.0, 0.0, 0.0)]).tolist()
+    [[1.0, 0.0, 0.0], [-1.0, 0.0, 0.0]]
 
-    # Round bvals by bmag.
-    bval_mag_norm = round_bvals(bvals, bmag=bmag)
+    >>> affine = np.eye(4)
+    >>> affine[0, 0] *= -1.0  # Make it LAS
```

Comment:
Instead of thinking on all the possible combinations of all possible inputs, just write tests where you know the answer. You'll save lots of time.